x-pack/plugins/ml/server/saved_objects/saved_objects.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SavedObjectsServiceSetup } from 'kibana/server';
+import mappings from './mappings.json';
+
+import { migrations } from './migrations';
+
+export const ML_SAVED_OBJECT_TYPE = 'ml-job';
+
+export function setupSavedObjects(savedObjects: SavedObjectsServiceSetup) {
+  savedObjects.registerType({
+    name: ML_SAVED_OBJECT_TYPE,
+    hidden: false,
```

Comment:
Switched to hidden saved object in https://github.com/elastic/kibana/pull/77916/commits/8fd05f3807335d192b4d1a7b9d8a5669c7f021b4

This also replaces the server side license class with a route guard class. As the required functionality had grown outside of the scope of just license checking.

Reply:
After making this change, i have realised that it has broken how we share our functions to other plugins.
We require that the consuming plugin supply the saved object client, this will be a normal client which cannot see the hidden ml job objects.
We also require the plugin supplied the `request` object, so I could in theory change it so we scope our saved object client based on this request.
However, plugins which are calling our functions from the alerting framework do not have the ability to get a real `request` object, so we allow a "fake" request object and just use the internal user for making calls to es.
https://github.com/elastic/kibana/issues/64588
I think until this is fixed, the ml saved objects cannot be `hidden`.

Reply:
I've changed the saved object so it is no longer hidden https://github.com/elastic/kibana/pull/77916/commits/807cd28b2d9b4f2a0b42e07046b7cac50302ac36
I'll keep in the other changes where we scope the saved object client for normal requests, so we can change this back once https://github.com/elastic/kibana/issues/64588 is fixed.